backend/inspirehep/orcid/domain_models.py

```diff
@@ -153,10 +153,9 @@ def push(self):
             # of the clashing work in ORCID and push a fresh version of that
             # record.
             # This scenario might be triggered by a merge of 2 records in Inspire.
-            if self.pushing_duplicated_identifier:
-                raise exceptions.DuplicatedExternalIdentifierPusherException
-            self._push_work_with_clashing_identifier()
-            putcode = self._post_or_put_work(putcode)
+            if not self.pushing_duplicated_identifier:
+                self._push_work_with_clashing_identifier()
+            raise exceptions.DuplicatedExternalIdentifierPusherException
```

Comment:
maybe add a comment here to say this causes it to be retried in the celery task?